### PR TITLE
DD Lambda Extension support on `lambda instrument` command

### PR DESCRIPTION
**Pull request by zARODz11z**
### What and why?
This PR introduces support for the DD lambda library extension to the `datadog-ci lambda instrument` command. The DD Lambda Extension is a special version of the Datadog Agent that the Serverless & Agent-Core team are working on releasing to GA, therefore the Serverless team would like all our instrumentation integrations to have support for the DD Lambda Extension.
Summary of changes:
* Addition of DD Lambda Extension via a layer
* Addition of DD_API_KEY/DD_API_KMS_KEY environment variables on lambda configs
* Block extension & forwarder to be included at the same time
* Make sure the the lambda function config already has a DD_API_KEY/DD_KMS_API_KEY or is given an api key through the users exported environment variables. Making it so the api keys never have to be placed in a config file, they can just be exported in the users environment.
### How?
The DD Lambda Extension is added to a lambda function as a layer therefore I followed the convention of how the DD Lambda Library layers were added with a few code changes to allow the extension to be added.

### Testing
I manually tested my changes by running the datadog-ci lambda instrument command on a sample lambda function in sandbox and confirmed the layers (lambda library and extension) were added. 
These are the cases I manually tested: 
- [x] Forwarder with lambda library layer
- [x] Extension and Forwarder at same time (correctly returns a 1 exit code)
- [x] ApiKey & ApiKMS key (always added if given and not already present on lambda config)
- [x] Extension with lambda library (eg. no problem adding a node lambda library layer with extension)
- [x] Extension with DD_API_KEY & DD_KMS_API_KEY exported in environment
- [x] Extension with DD_KMS_API_KEY exported in environment
- [x] Extension with DD_API_KEY exported in environment
- [x] Extension with no apiKey given or already on lambda config(correctly throws an error)
### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

**Comment by zARODz11z** [comment]
p.s. I've read the release [process](https://github.com/DataDog/datadog-ci#release-process) for the datadog-ci, i'm waiting to update the version using `yarn version` until I get an approval on my PR

**Comment by zARODz11z** [comment]
Github actions are degraded ATM https://www.githubstatus.com/#

File: src/commands/lambda/function.ts
Comment (DarcyRaynerDD) [inline src/commands/lambda/function.ts:191]: nit: seems like most of this logic is duplicated from line 183. Could it be moved into a function?

**Review by DarcyRaynerDD — APPROVED** [review]
Nice work, LGTM overall

File: src/commands/lambda/function.ts
Comment (zARODz11z) [reply src/commands/lambda/function.ts:207]: yeah I agree that its crowding the function, I moved it to `determineIfLayerShouldBeAdded`

File: src/commands/lambda/function.ts
Comment (DarcyRaynerDD) [inline src/commands/lambda/function.ts:236]: The name is kind of confusing given the return type. I would expect it to return a boolean based on that name. Maybe just addLayerARN? I think you could also keep the extension/env var check in calculateUpdateRequest, since it's not related to what the function indicates it's doing.

**Review by DarcyRaynerDD — APPROVED** [review]
LGTM overall, had just 1 nit

File: src/commands/lambda/function.ts
Comment (zARODz11z) [reply src/commands/lambda/function.ts:236]: yeah I was a bit iffy on the name, thanks for pointing it out. I also moved the extension stuff back into calculateUpdateRequest

**Pull request merged**
